src/string_piece.h

```diff
@@ -21,51 +21,57 @@ using namespace std;
 
 #include <string.h>
 
-/// StringPiece represents a slice of a string whose memory is managed
-/// externally.  It is useful for reducing the number of std::strings
+/// BasicStringPiece represents a slice of a string whose memory is managed
+/// externally.  It is useful for reducing the number of strings
 /// we need to allocate.
-struct StringPiece {
-  typedef const char* const_iterator;
+template <class StringType, class CharType>
```

Comment:
Don't take a template parameter for StringType.

Take std::basic_string

http://www.cplusplus.com/reference/string/basic_string/

E.g the constructor becomes BasicStringPiece(std::basic_string<CharType> const& str);

Reply:
@jlonnberg